api/lua/definition/poptracker.lua

```diff
@@ -0,0 +1,548 @@
+--
+
+---- Helpers ----
+
+---@alias AnyObject JsonItem | LuaItem | LocationSection | Location
+
+
+---- Globals ----
+
+---Currently running PopTracker version as string.
+---@type string
+PopVersion = "0.25.7"
+
+---Set to true to get more error or debug output.
+---@type boolean
+DEBUG = false
+
+---If defined in global scope, called when a game is connected (currently memory interfaces only)
+---@type function():nil
+autotracker_started = nil
+
+---If defined in global scope, called when auto-tracking is disabled by the user (currently memory interfaces only)
+autotracker_stopped = nil
+
+
+---- Tracker ----
+
+---@class Tracker
+Tracker = {}
+
+---@type string
+Tracker.ActiveVariantUID = ""
+
+---load items from json
+---@param jsonFIlename string
+---@return boolean true on success
+function Tracker:AddItems(jsonFIlename) end
+
+---load maps from json
+---@param jsonFilename string
+---@return boolean true on success
+function Tracker:AddMaps(jsonFilename) end
+
+---load locations from json
+---@param jsonFilename string
+---@return boolean true on success
+function Tracker:AddLocations(jsonFilename) end
+
+---load layouts from json
+---@param jsonFilename string
+---@return boolean true on success
+function Tracker:AddLayouts(jsonFilename) end
+
+---Returns number of items that provide the code (sum of count for consumables).
+---@param code string
+---@return integer
+function Tracker:ProviderCountForCode(code) end
+
+---Returns item for `code` or location section for `@location/section`
+---@param code string
+---@return AnyObject?
+function Tracker:FindObjectForCode(code) end
+
+---Sends a hint to the UI, see https://github.com/black-sliver/PopTracker/blob/master/doc/PACKS.md#ui-hints
+---Only available in PopTracker, since 0.11.0
+---@param name string
+---@param value string
+---@return nil
+function Tracker:UiHint(name, value) end
+
+---can be set to true from Lua to pause running logic rules.
+---@type boolean
+Tracker.BulkUpdate = false
+
+
+---- ScriptHost ----
+
+---@class ScriptHost
+ScriptHost = {}
+
+---Load and execute a Lua script from absolute filename inside pack.
+---* sets `...` variable to mod name (for relative require) since 0.25.6
+---* require can be used instead since 0.25.6 (0.21.0 in some cases)
+---    * `require "foo.baz"` will try `/scripts/foo/baz.lua`, `/scripts/foo/baz/init.lua`,
+---    `/foo/baz.lua` and `/foo/baz/init.lua`
+---    * LoadScript needs absolute path instead
+---    * LoadScript needs to use globals, require can return for `local x = require "foo"`
+---* require is recommended if backwards compatibility is not required.
+---@param luaFilename string
+---@return boolean true on success
+function ScriptHost:LoadScript(luaFilename) end
+
+---add a memory watch for auto-tracking, see [AUTOTRACKING.md](https://github.com/black-sliver/PopTracker/blob/master/doc/AUTOTRACKING.md)
+---@param name string
+---@param addr integer
+---@param len integer
+---@param callback function(segment:Segment):nil
```

Comment:
```suggestion
---@param callback fun(segment:Segment):nil
```
function breaks the display in the VS Code addon
![image](https://github.com/black-sliver/PopTracker/assets/20642139/eb603fc2-faea-4517-b2ba-e410ecc7856e)

Reply:
This is something i mentioned on stream: `function` seems to be the old syntax (EmmyLua) and `fun` seems to be the new syntax (latest LuaCATS), however the new syntax is not complete yet (the issue is still open). But you are probably right that we should use `fun` already. I'm gonna do that in a bit.